azurerm/resource_arm_monitor_metric_alert.go

```diff
@@ -0,0 +1,450 @@
+		d.Set("frequency", alert.EvaluationFrequency)
+		d.Set("window_size", alert.WindowSize)
+		if err := d.Set("target_ids", flattenMonitorMetricAlertStringArray(alert.Scopes)); err != nil {
+			return err
+		}
+		if err := d.Set("criteria", flattenMonitorMetricAlertCriteria(alert.Criteria)); err != nil {
+			return err
+		}
+		if err := d.Set("action", flattenMonitorMetricAlertAction(alert.Actions)); err != nil {
+			return err
+		}
+	}
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmMonitorMetricAlertDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).monitorMetricAlertsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["metricAlerts"]
+
+	if resp, err := client.Delete(ctx, resGroup, name); err != nil {
+		if !response.WasNotFound(resp.Response) {
+			return fmt.Errorf("Error deleting metric alert %q (resource group %q): %+v", name, resGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandMonitorMetricAlertStringArray(v []interface{}) *[]string {
+	result := make([]string, 0)
+	for _, item := range v {
+		result = append(result, item.(string))
+	}
+	return &result
+}
+
+func expandMonitorMetricAlertCriteria(v []interface{}) *insights.MetricAlertSingleResourceMultipleMetricCriteria {
+	criterias := make([]insights.MetricCriteria, 0)
+	for i, criteriaValue := range v {
+		val := criteriaValue.(map[string]interface{})
+
+		mNS := val["metric_namespace"].(string)
+		mName := val["metric_name"].(string)
+		aggr := val["aggregation"].(string)
+		op := val["operator"].(string)
+		threshold := val["threshold"].(float64)
+
+		dimensions := make([]insights.MetricDimension, 0)
+		for _, dimension := range val["dimension"].([]interface{}) {
+			dVal := dimension.(map[string]interface{})
+			dName := dVal["name"].(string)
+			dOp := dVal["operator"].(string)
+			dValuesRaw := dVal["values"].([]interface{})
+			dimensions = append(dimensions, insights.MetricDimension{
+				Name:     utils.String(dName),
+				Operator: utils.String(dOp),
+				Values:   expandMonitorMetricAlertStringArray(dValuesRaw),
+			})
+		}
+
+		criterias = append(criterias, insights.MetricCriteria{
+			Name:            utils.String(fmt.Sprintf("Metric%d", i+1)),
+			MetricNamespace: utils.String(mNS),
+			MetricName:      utils.String(mName),
+			TimeAggregation: aggr,
+			Operator:        op,
+			Threshold:       utils.Float(threshold),
+			Dimensions:      &dimensions,
+		})
+	}
+	return &insights.MetricAlertSingleResourceMultipleMetricCriteria{
+		AllOf:     &criterias,
+		OdataType: insights.OdataTypeMicrosoftAzureMonitorSingleResourceMultipleMetricCriteria,
+	}
+}
+
+func expandMonitorMetricAlertAction(v []interface{}) *[]insights.MetricAlertAction {
+	actions := make([]insights.MetricAlertAction, 0)
+	for _, actionValue := range v {
+		val := actionValue.(map[string]interface{})
+
```

Comment:
might i suggest the following whitespace:

```go
val := actionValue.(map[string]interface{})
agID := val["action_group_id"].(string)

props := make(map[string]*string)
if propsValue, ok := val["webhook_properties"]; ok {
```

Reply:
This comment is out of date.